### DISCUSSION TITLE
Can we keep using on-device text recognition on newer rn firebase?

### DISCUSSION THREAD

**Discussion by flixyudh** [invertase/react-native-firebase#4794] [Q&A] [answered]
since the docs of [Text Recognition (TR)](https://rnfirebase.io/ml/text-recognition) only provide cloud TR,
i'm curious how can i keep using on-device TR, since cloud TR need me to use billing account and enabling API
does on-deivce TR still exist on this library?

I've found this (*_Old docs_*](https://rnfb-docs.netlify.app/ml-vision/text-recognition#on-device-text-recognition)
and when i tried this code
```
import vision from '@react-native-firebase/ml-vision';

async processDocument(localPath) {
    const processed = await vision().textRecognizerProcessImage(localPath.path);
  }
```

i got an error
```
Possible Unhandled Promise Rejection (id: 1):
TypeError: (0, _mlVision.default) is not a function. (In '(0, _mlVision.default)()', '(0, _mlVision.default)' is undefined)
processDocument$
```

with detail `package.json`:
```
"dependencies": {
    "@react-native-firebase/analytics": "^10.3.1",
    "@react-native-firebase/app": "^10.3.0",
    "@react-native-firebase/messaging": "^10.3.1",
    "@react-native-firebase/ml": "^10.3.1",
    "@react-native-firebase/ml-vision": "^7.4.13",
    ....
}

**Comment by mikehardy** [comment, marked as answer]
No, you cannot - related/duplicate - follow here: https://github.com/invertase/react-native-firebase/discussions/4555
